### PR TITLE
blender: update to 2.83.3, enable alembic support

### LINKED ISSUE #23112: Please add alembic to Blender
Hi all,
I've tried to add alembic to the blender template but there is an error (i'm unskilled c++ programmer ;) ).
I've tried with direct blender compilation from git (with 2.83 version) with same error.
```
[ 84%] Building CXX object source/blender/io/alembic/CMakeFiles/bf_alembic.dir/intern/abc_reader_mesh.cc.o
/home/bda/src/blender/source/blender/io/alembic/intern/abc_reader_mesh.cc: In function 'void read_mverts(CDStreamConfig&, const AbcMeshData&)':
/home/bda/src/blender/source/blender/io/alembic/intern/abc_reader_mesh.cc:162:41: error: no matching function for call to 'read_mverts(MVert*&, const P3fArraySamplePtr&, std::nullptr_t)'
  162 |   read_mverts(mverts, positions, nullptr);
      |                                         ^
In file included from /home/bda/src/blender/source/blender/io/alembic/intern/abc_reader_mesh.cc:21:
/home/bda/src/blender/source/blender/io/alembic/intern/abc_reader_mesh.h:80:6: note: candidate: 'void read_mverts(MVert*, Alembic::Abc::v12::P3fArraySamplePtr, Alembic::Abc::v12::N3fArraySamplePtr)'
   80 | void read_mverts(MVert *mverts,
      |      ^~~~~~~~~~~
/home/bda/src/blender/source/blender/io/alembic/intern/abc_reader_mesh.h:82:60: note:   no known conversion for argument 3 from 'std::nullptr_t' to 'Alembic::Abc::v12::N3fArraySamplePtr' {aka 'std::tr1::shared_ptr<Alembic::Abc::v12::TypedArraySample<Alembic::Abc::v12::N3fTPTraits> >'}
   82 |                  const Alembic::AbcGeom::N3fArraySamplePtr normals);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
/home/bda/src/blender/source/blender/io/alembic/intern/abc_reader_mesh.cc:151:13: note: candidate: 'void read_mverts(CDStreamConfig&, const AbcMeshData&)'
  151 | static void read_mverts(CDStreamConfig &config, const AbcMeshData &mesh_data)
      |             ^~~~~~~~~~~
/home/bda/src/blender/source/blender/io/alembic/intern/abc_reader_mesh.cc:151:13: note:   candidate expects 2 arguments, 3 provided
make[2]: *** [source/blender/io/alembic/CMakeFiles/bf_alembic.dir/build.make:148 : source/blender/io/alembic/CMakeFiles/bf_alembic.dir/intern/abc_reader_mesh.cc.o] Erreur 1
make[1]: *** [CMakeFiles/Makefile2:6238 : source/blender/io/alembic/CMakeFiles/bf_alembic.dir/all] Erreur 2
make: *** [Makefile:183 : all] Erreur 2
```
Apparently alembic compile well on archlinux without any patch.

Thanks

### PR DESCRIPTION
**Pull request by mvf**
Smoke-tested (export and import simple mesh) on x86_64. Fixes #23112

### CODE REVIEW

**Comment by fosslinux** [comment]
I think the TR1 in alembic would be better as a build-option.

**Comment by q66** [comment]
what's up with that tr1 change

**Comment by mvf** [comment]
**TL;DR**
- the TR1 flavor of `alembic` is useless
- `libatomic` is required on `armv6l*`

**Prose**

> what's up with that tr1 change

TR1 is an obsolete non-standard C++ extension that has been effectively superseded by C++11. `alembic`'s TR1 build option is merely a fallback for people stuck on GCC < 4.8. Enabling it changes `alembic`'s interfaces, making it useless on modern systems. That's also why a build option makes no sense. From the README:
```
Some examples of OPTIONS you may want or need to use include:
[...]
-DALEMBIC_LIB_USES_TR1=ON or -DALEMBIC_LIB_USES_BOOST=ON if you do not have
a C++11 capable compiler specify one of these to use TR1, or boost as
a dependency of the Alembic library.
```
My guess is that the original `alembic` packager wasn't a C++ person and thought enabling TR1 was a neat way to get it to link without patching. From `lib/Alembic/CMakeLists.txt`:
```cmake
# link in atomic if we are using tr1 and the compiler is new enough
# to want  __atomic_compare_exchange_n
IF ( ${ALEMBIC_LIB_USES_TR1} AND CMAKE_COMPILER_IS_GNUCXX AND
    CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 4.8 )
    TARGET_LINK_LIBRARIES( Alembic atomic )
ENDIF()
```
> without tr1, atomic shouldn't be necessary at all

`armv6l*` disagrees:
```
/usr/lib/gcc/arm-linux-gnueabihf/9.3.0/../../../../arm-linux-gnueabihf/bin/ld: ../../libAlembic.so.1.7.12: undefined reference to `__atomic_fetch_or_8'
/usr/lib/gcc/arm-linux-gnueabihf/9.3.0/../../../../arm-linux-gnueabihf/bin/ld: ../../libAlembic.so.1.7.12: undefined reference to `__atomic_compare_exchange_8'
/usr/lib/gcc/arm-linux-gnueabihf/9.3.0/../../../../arm-linux-gnueabihf/bin/ld: ../../libAlembic.so.1.7.12: undefined reference to `__atomic_load_8'
/usr/lib/gcc/arm-linux-gnueabihf/9.3.0/../../../../arm-linux-gnueabihf/bin/ld: ../../libAlembic.so.1.7.12: undefined reference to `__atomic_store_8'
```

**Comment by mvf** [comment]
Guess I didn't really explain the "useless" part. It means: Recent applications don't compile against the TR1 flavor of `alembic`: https://github.com/void-linux/void-packages/issues/23112#issue-642559341

**Comment by fosslinux** [comment]
Well, this is the most well-thought out answer to a question I have ever given.

I get it now, thanks, agree with the changes in that case.

**Comment by q66** [comment]
that's strange, i only see atomic ever used if TR1 is on https://github.com/alembic/alembic/search?q=atomic&unscoped_q=atomic

but if that's the case, leave that in place, yeah - it'll be necessary for 32-bit powerpc and mips in the same way

**Comment by mvf** [comment]
Atomics are always used. It won't even compile without. From [lib/Alembic/AbcCoreOgawa/StreamManager.cpp](https://github.com/alembic/alembic/blob/HEAD/lib/Alembic/AbcCoreOgawa/StreamManager.cpp):
```c++
// C++11 std::atomics version
#if !defined( ALEMBIC_LIB_USES_TR1 ) && __cplusplus >= 201103L
#define COMPARE_EXCHANGE( V, COMP, EXCH ) V.compare_exchange_weak( COMP, EXCH, std::memory_order_seq_cst, std::memory_order_seq_cst )
...
// gcc 4.8 and above not using C++11
#elif defined(__GNUC__) && __GNUC__ >= 4 && __GNUC_MINOR__ >= 8
#define COMPARE_EXCHANGE( V, COMP, EXCH ) __atomic_compare_exchange_n( &V, &COMP, EXCH, false, __ATOMIC_SEQ_CST, __ATOMIC_SEQ_CST )
// gcc 4.1 and above not using C++11
#elif defined(__GNUC__) && __GNUC__ >= 4 && __GNUC_MINOR__ >= 1
#define COMPARE_EXCHANGE( V, COMP, EXCH ) __sync_bool_compare_and_swap( &V, COMP, EXCH )
#else
#error Please contact alembic-discuss@googlegroups.com for support.
#endif
```

**Comment by q66** [comment]
silly me, it uses `!` (and intrinsics as a fallback)

yes, lgtm

**Comment by mvf** [comment]
@Gottox Since you're the `blender` maintainer, care to merge this? The `x86_64` build timed out, but I actually tested it.

**Comment by Gottox** [comment]
Thanks for taking care! Looks good to me! :+1:
